service/matching/taskQueueManager.go

```diff
@@ -474,42 +478,42 @@ func (c *taskQueueManagerImpl) MutateVersioningData(ctx context.Context, mutator
 	if err != nil {
 		return err
 	}
+	if !c.taskQueueID.IsRoot() {
+		return nil
+	}
 	// If this is the root partition, notify partitions that they should fetch changed data from us.
-	if c.taskQueueID.IsRoot() {
-		numParts := util.Max(c.config.NumReadPartitions(), c.config.NumWritePartitions())
-		wg := &sync.WaitGroup{}
-		for i := 1; i < numParts; i++ {
-			wg.Add(1)
-			go func(i int) {
-				tq := c.taskQueueID.mkName(i)
-				_, err := c.matchingClient.InvalidateTaskQueueMetadata(ctx,
-					&matchingservice.InvalidateTaskQueueMetadataRequest{
-						NamespaceId:    c.taskQueueID.namespaceID.String(),
-						TaskQueue:      tq,
-						VersioningData: newDat,
-					})
-				if err != nil {
-					c.logger.Warn("Failed to notify sub-partition of invalidated versioning data",
-						tag.WorkflowTaskQueueName(tq), tag.Error(err))
-				}
-				wg.Done()
-			}(i)
-			wg.Wait()
-		}
+	numParts := util.Max(c.config.NumReadPartitions(), c.config.NumWritePartitions())
+	wg := &sync.WaitGroup{}
+	for i := 1; i < numParts; i++ {
+		wg.Add(1)
+		go func(i int) {
+			tq := c.taskQueueID.mkName(i)
+			_, err := c.matchingClient.InvalidateTaskQueueMetadata(ctx,
```

Comment:
I think you have to loop over workflow + activity here? oh, but then how do you direct the request to the activity queues? pretty sure we have to add tqt to protos actually, otherwise it'll break after #3202